packages/main/src/Link.ts

```diff
@@ -260,6 +306,32 @@ class Link extends UI5Element implements ITabbable {
 			&& this._dummyAnchor.protocol === loc.protocol);
 	}
 
+	_ensureNoIconOnlyLink() {
+		if (this._isIconOnly) {
+			throw new Error("Icon only links are not supported.");
+		}
+	}
+
+	get _isIconOnly() {
+		return (this.icon || this.endIcon) && !this.textContent?.trim();
+	}
+
+	get iconMode() {
```

Comment:
iconMode is guaranteed to be called only if the `icon` is present and `ui5-icon` is about to be rendered. And you can directly return the required IconMode.Decorative without these checks. (Same for the endIconMode)

```hbs
{{#if icon}}
		<ui5-icon
			class="ui5-link-icon"
			name="{{icon}}"
			mode="{{iconMode}}"
			part="icon">
		</ui5-icon>
	{{/if}}
```